direct.mk

```diff
@@ -139,14 +135,19 @@ help: ## this help
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
 
 .PHONY: dep-validate
-dep-validate:
+dep-validate: go-mod-vendor
 	@echo "+ $@"
-	$(if $(VNDR), , \
-		$(error Please install vndr: go get github.com/lk4d4/vndr))
-	@rm -Rf .vendor.bak
-	@mv vendor .vendor.bak
-	@$(VNDR)
-	@test -z "$$(diff -r vendor .vendor.bak 2>&1 | tee /dev/stderr)" || \
-		(echo >&2 "+ inconsistent dependencies! what you have in vendor.conf does not match with what you have in vendor" && false)
-	@rm -Rf vendor
-	@mv .vendor.bak vendor
+	@go mod tidy -modfile vendor.mod
```

Comment:
@thaJeztah @crazy-max 👁️

Reply:
🙈 should this one be under the `go-mod-vendor:` target?

Reply:
LGTM

@thaJeztah We should only `mod tidy` on validation. See https://github.com/docker/cli/pull/3387#discussion_r768766784

Reply:
Yeah, as @crazy-max said - currently the idea is to detect untidy cases in validation stage, tidying as such should be up to the contributor. We could change this in the future, just that initially it seems reasonable to assume that `go mod tidy` has too much of a side-effect as it can rewrite `vendor.sum` entirely.